### ISSUE TITLE
disabling a single line effects multiple lines?

### ISSUE THREAD

**Issue by austin987** [koalaman/shellcheck#960]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC 2039
- My shellcheck version (`shellcheck --version` or "online"): online
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

# shell check catches this:
# shellcheck disable=SC2039
echo -e foo

# but misses this:
a="$(echo -e foo)"
echo "$a"

```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
SC2039: In POSIX sh, echo flags are undefined.

If i reenable the check, I get warnings for both. Disabling it for the first disables for the next occurence as well, even though it is not the start of a function block.

**Comment by koalaman** [comment]
This is an unfortunate compromise from allowing file-wide directives right after the shebang, while also allowing them right before commands.

Here's the description from [the wiki](https://github.com/koalaman/shellcheck/wiki/Directive):

>Directives instead of or immediately after the shebang apply to the entire script. Otherwise, they are scoped to the structure that follows it (such as all branches of a `case` statement, or an entire function).

>There is no support for scoping a directive to the first structure of the script. In these cases, use a dummy command `true` or `:` and then add directives, such as

    # This directive applies to the entire script
    # shellcheck disable=2086
    true
    # This directive only applies to this function
    # shellcheck disable=2043
    f() {
      ...
    }

**Comment by austin987** [comment]
..oops, thanks for the correction :)

**Comment by marcospgp** [comment]
No way to support something like `// eslint-disable-next-line`?

**Comment by brother** [comment]
> No way to support something like `// eslint-disable-next-line`?

Please open a new issue and describe the wish further. 
This issue was closed almost six years ago and the engine and possibilities have probably changed in the mean time. Add a link to this issue for reference if you like.

**Comment by marcospgp** [comment]
@brother Thanks for the reply!

Actually it was not super clear but upon re-researching this I realized that `shellcheck-disable` works as expected.

See "Ignoring one specific instance in a file" here: https://www.shellcheck.net/wiki/Ignore

For those arriving here from Google, as I did :)
